File: main/java/com/university/demo/entity/Song.java
package com.university.demo.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.activerecord.Model;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * Song2
 * @author
 * @since 2022年10月09日08:22:45
 */

@TableName("tb_song2")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Song extends Model<Song> {

    private static final long serialVersionUID=1L;

    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;
    @TableField(value = "songId")
    private String songId;
    @TableField(value = "songName")
    private String songName;
    private String alia;
    private String pic;
    @TableField(value = "singerId")
    private String singerId;
    @TableField(value = "singerName")
    private String singerName;
    @TableField(value = "albumId")
    private String albumId;
    @TableField(value = "albumName")
    private String albumName;
    private String dt;
    private String pop;
    private String fee;
    private String mv;
    private String cd;
    private String no;
    @TableField(value = "originCoverType")
    private String originCoverType;
    @TableField(value = "publishTime")
    private LocalDateTime publishTime;
    private String mvUrl;
    @Override
    protected Serializable pkVal() {
        return this.id;
    }
}

File: main/java/com/university/demo/python/pythonscript/song.py
# -*- codeing = utf-8 -*-
# Author:  Tesla Tech
# QQ: 1693353672
# @Time :2022年10月2日22:29:01
# @Author: Administrator
# @File :song.py
from flask_marshmallow import Marshmallow
from flask_sqlalchemy import SQLAlchemy
db = SQLAlchemy()

ma = Marshmallow()

# 歌曲的表
class Song(db.Model):
    __tablename__ = 'tb_song2'  # 表名
    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
    songId = db.Column(db.String(255))
    songName = db.Column(db.String(255))
    alia = db.Column(db.String(255))
    pic = db.Column(db.String(255))
    singerName = db.Column(db.String(255))
    albumName = db.Column(db.String(255))
    dt = db.Column(db.String(50))
    pop = db.Column(db.String(20))
    originCoverType = db.Column(db.String(20))
    fee = db.Column(db.String(50))
    mv_url = db.Column(db.TEXT)
    mv = db.Column(db.String(50))
    publishTime = db.Column(db.DATETIME)  #时间的处理

class songSchema(ma.Schema):
    class Meta:
        fields = ('id','songId', 'songName', 'alia', 'pic', 'singerName', 'albumName', 'dt', 'pop', 'publishTime','originCoverType','fee',
                  'mv_url','mv')

song_schema = songSchema()
song_schema = songSchema(many=True)

class RankSchema(ma.Schema):
    class Meta:
        fields = ('name', 'value')

rank_schema = RankSchema()
rank_schema = RankSchema(many=True)




